### ISSUE TITLE
Login: "Text me a code instead" sometimes doesn't result in an SMS with a verification code when using Google login plus 2FA

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/WordPress-iOS#13807]
Steps to reproduce:

1. Start with an account with Google sign-in and two-factor authentication enabled.
1. Tap "Log in".
1. Tap "Continue with Google".
1. Tap "Text me a code instead".
1. Wait for an SMS.
1. If an SMS doesn't arrive, try tapping "Text me a code instead" again.
1. If you are able to log in successfully, log out and try the steps again a few times.

**Result:** sometimes tapping "Text me a code instead" doesn't send a code or the code isn't received quickly enough, this happens most often after logging in using SMS successfully just prior to trying it a second time. ([1m52s](https://cloudup.com/cLsgcdSJUGR), [2m45s](https://cloudup.com/ckM1hTn0ZMp))

<img src="https://user-images.githubusercontent.com/1119271/78308786-a3685a00-7506-11ea-822f-c7bf2b8a1514.PNG" width="200" alt="IMG_3715">
<sup>Tested with WPiOS 14.5.0.1 TestFlight beta on iPhone 11 iOS 13.4.</sup>

**Comment by mindgraffiti** [comment]
Hi @designsimply! 

If I request a texted code within 5 minutes of a previous successful request, it fails. You can see the failure reported in the Activity Log: `Failed to request one time code`. I don't know if this is a limitation of our API, our servers, or something else. We'll check into it.

Added to UL&S backlog.

**Comment by rachelmcr** [comment]
I think we're not handling the `sms_code_throttled` error for 2FA correctly in all cases.

I ran into this while testing today using Google login on an account with SMS 2FA (the account isn't set up with an authenticator app at all). I started with these steps:

1. Start with an account with Google sign-in and two-factor authentication via SMS enabled.
2. Tap "Log in."
3. Tap "Continue with Google."
4. Select my Google account.
5. Notice that I am on the 2FA screen and I get an SMS with my 2FA code.
6. Instead of continuing to log in, go back to the main login screen and repeat steps 1-4 immediately.

Result: I get a screen with the error message `You can request a code via SMS once per minute. Please wait and try again." My activity log shows the `sms_code_throttled` error with that same error message.

<img src="https://user-images.githubusercontent.com/8658164/78891512-99db7680-7a5f-11ea-8c8b-fdf3f5214faf.PNG" width="50%">

However, login silently fails if I then go back and try these steps:

1. Start with the same account with Google sign-in and two-factor authentication via SMS enabled.
2. Tap "Log in."
3. Tap "Continue with WordPress.com."
4. Enter my Google email address.
5. Enter my account password.
6. Wait for the SMS with my 2FA code or tap "Text me a code instead." Notice that I never get the 2FA code and no error appears on screen (same behavior as described in the original issue).

**Comment by designsimply** [comment]
Agreed. I don't think the error handling takes into account logins that happen very closely together.

Here's a case where the verification code didn't work at all for me and an SMS did not get sent despite the "SMS Sent" message appearing after I tapped the "Text me a code instead" option when the auth code didn't work. It happened within a minute after I did the test noted at https://github.com/wordpress-mobile/WordPress-iOS/issues/13806#issuecomment-612235896.

Video: [1m34s](https://cloudup.com/c_Md-rVIjaI)
App logs: [2fa-verification-code-fails-and-sms-send-fails-wpios-14.6-beta-13807.txt](https://github.com/wordpress-mobile/WordPress-iOS/files/4463495/app-logs-after-2fa-verification-code-fails-and-sms-send-fails-wpios-14.6-beta-13807.txt)
<p style="clear:both;"><sup>Tested with WPiOS 14.6.0.1 TestFlight beta on iPhone 6S iOS 13.3.1.</sup></p>

**Comment by mindgraffiti** [comment]
Did a lot of research on the client side and couldn't come up with a reasonable work around. In the LoginFacade, there is [an error being logged ](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/Services/LoginFacade.swift#L21)but not handled. The reason why it isn't handled, is because the [API request frequently reports a failure](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/develop/WordPressKit/WordPressComOAuthClient.swift#L189), even though a text message was successfully sent. So sending the user the error message would be confusing, because the requested OTP SMS was received. 

This issue needs researched on the API side and a bug filed there.

**Comment by mindgraffiti** [comment]
Moving to Groundskeeping, as the UL&S project is wrapping up.

**Comment by jkmassel** [comment]
This is no longer applicable now that we use the WP.com OAuth flow.
